libs/storage/Tsavorite/cs/src/core/ClientSession/LockableContext.cs

```diff
@@ -65,9 +67,14 @@ internal LockableContext(ClientSession<Key, Value, Input, Output, Context, Funct
             // This is the same as DoInternalTryLock but without timeout; it will keep trying until it acquires all locks.
             var end = start + count - 1;
 
+            int retryCount = 0;
         Retry:
-            long prevBucketIndex = -1;
+            if (retryCount >= KeyLockMaxRetryAttempts)
+            {
+                return false;
+            }
```

Comment:
let's move this down to line 95, where you update retryCount. That way we don't pay the "if" cost on the first iteration